File: public/UserDao.php
<?php

class UserDao
{
  public $dsn;
  public $user;
  public $password;

  function __construct() {
    $this->dsn = getenv('DSN');
    $this->user = getenv('DBUSER');
    $this->password = getenv('DBPASS');
  }

  function get($squadId) {
    try {
      $dbh = new PDO($this->dsn, $this->user, $this->password);

      $sql = "SELECT * FROM users WHERE squad_id='".$squadId."'";
      $stmt = $dbh->query($sql);
      return $stmt;

    } catch (PDOException $e) {
      echo 'データベースにアクセスできません！' . $e->getMessage();
    } finally {
      $dbh = null; //close
    }
  }

  function post($userId, $userName, $squadId) {
    try {
      $dbh = new PDO($this->dsn, $this->user, $this->password);

      $sql = "INSERT INTO users (user_id, user_name, squad_id) VALUES ("
            . "'" . $userId . "', "
            . "'" . $userName . "', "
            . "'" . $squadId . "'"
            .")";
      $stmt = $dbh->query($sql);

    } catch (PDOException $e) {
      echo 'データベースにアクセスできません！' . $e->getMessage();
    } finally {
      $dbh = null; //close
    }
  }

  function deleteAllBySquadId($squadId) {
    try {
      $dbh = new PDO($this->dsn, $this->user, $this->password);

      $sql = "DELETE FROM users WHERE squad_id='{$squadId}'";
      $stmt = $dbh->query($sql);

    } catch (PDOException $e) {
      echo 'データベースにアクセスできません！' . $e->getMessage();
    } finally {
      $dbh = null; //close
    }
  }

  function deleteAll() {
    try {
      $dbh = new PDO($this->dsn, $this->user, $this->password);

      $sql = "DELETE FROM users";
      $stmt = $dbh->query($sql);

      $sql = "SELECT setval ('users_id_seq', 1, false)";
      $stmt = $dbh->query($sql);

    } catch (PDOException $e) {
      echo 'データベースにアクセスできません！' . $e->getMessage();
    } finally {
      $dbh = null; //close
    }
  }
}

File: userDao.rb
require 'pg'
require_relative './User'
require_relative './UserList'

class UserDao
    def get_all
        userList = []
        conn = PG::Connection.new(ENV["DATABASE_URL"])
        conn.exec("SELECT * FROM users") do |result|
            result.each do |row|
                userList.push User.new(row["user_id"], row["user_name"], row["squad_id"])
            end
        end
        return UserList.new(userList)
    end

    def get squadId
        userList = []
        conn = PG::Connection.new(ENV["DATABASE_URL"])
        conn.exec("SELECT * FROM users WHERE squad_id='#{squadId}'") do |result|
            result.each do |row|
                userList.push User.new(row["user_id"], row["user_name"], row["squad_id"])
            end
        end
        return UserList.new(userList)
    end

    def post userId, userName, squadId
        conn = PG::Connection.new(ENV["DATABASE_URL"])
        conn.exec("INSERT INTO users (user_id, user_name, squad_id) VALUES ('#{userId}', '#{userName}', '#{squadId}')")
    end

    def deleteAllBySquadId squadId
        conn = PG::Connection.new(ENV["DATABASE_URL"])
        conn.exec("DELETE FROM users WHERE squad_id='#{squadId}'")
    end
end
